management_api_app/models/domain/resource.py

```diff
@@ -46,3 +46,12 @@ def is_enabled(self) -> bool:
         if "enabled" not in self.resourceTemplateParameters:
             return True     # default behavior is enabled = True
         return self.resourceTemplateParameters["enabled"] is True
+
+    def get_resource_request_message_payload(self, action: str) -> dict:
```

Comment:
should action be an enum - RequestAction as in the original merged code? so we can control what actions are possible